Use the C++ compiler to build C++ code by default.

Patch by Krzysiek Pawlik
http://bugs.gentoo.org/show_bug.cgi?id=87974

--- jam-2.5/Jambase
+++ jam-2.5/Jambase
@@ -533,7 +533,7 @@
 	ASFLAGS		?= ;
 	AWK		?= awk ;
 	BINDIR		?= /usr/local/bin ;
-	C++		?= cc ;
+	C++		?= c++ ;
 	C++FLAGS	?= ;
 	CC		?= cc ;
 	CCFLAGS		?= ;
@@ -1070,6 +1070,7 @@
 	    case .C :	C++ $(<) : $(>) ;
 	    case .cc :	C++ $(<) : $(>) ;
 	    case .cpp : C++ $(<) : $(>) ;
+	    case .cxx : C++ $(<) : $(>) ;
 	    case .f :	Fortran $(<) : $(>) ;
 	    case .l :	Cc $(<) : $(<:S=.c) ;
 			Lex $(<:S=.c) : $(>) ;
--- jam-2.5/jambase.c
+++ jam-2.5/jambase.c
@@ -325,7 +325,7 @@
 "ASFLAGS		?= ;\n",
 "AWK		?= awk ;\n",
 "BINDIR		?= /usr/local/bin ;\n",
-"C++		?= cc ;\n",
+"C++		?= c++ ;\n",
 "C++FLAGS	?= ;\n",
 "CC		?= cc ;\n",
 "CCFLAGS		?= ;\n",
@@ -658,6 +658,7 @@
 "case .C :	C++ $(<) : $(>) ;\n",
 "case .cc :	C++ $(<) : $(>) ;\n",
 "case .cpp : C++ $(<) : $(>) ;\n",
+"case .cxx : C++ $(<) : $(>) ;\n",
 "case .f :	Fortran $(<) : $(>) ;\n",
 "case .l :	Cc $(<) : $(<:S=.c) ;\n",
 "Lex $(<:S=.c) : $(>) ;\n",
